HeliPort/CredentialsManager.swift

```diff
@@ -18,19 +18,39 @@ final class CredentialsManager {
         keychain = Keychain(service: Bundle.main.bundleIdentifier!)
     }
 
-    func save(_ network: NetworkInfo, password: String) {
+    func save(_ network: NetworkInfo) {
+        guard let networkAuthJson = try? String(data: JSONEncoder().encode(network.auth), encoding: .utf8) else {
+            return
+        }
+        network.auth = NetworkAuth()
+        guard let networkInfoJson = try? String(data: JSONEncoder().encode(network), encoding: .utf8) else {
+            return
+        }
+
         Log.debug("Saving password for network \(network.ssid)")
-        keychain[string: network.keychainKey] = password
+        try? keychain.comment(networkInfoJson).set(networkAuthJson, key: network.keychainKey)
     }
 
-    func get(_ network: NetworkInfo) -> String? {
-        guard let password = keychain[string: network.keychainKey], !password.isEmpty else {
+    func get(_ network: NetworkInfo) -> NetworkAuth? {
+        guard let password = keychain[string: network.keychainKey],
+            let jsonData = password.data(using: .utf8) else {
             Log.debug("No stored password for network \(network.ssid)")
             return nil
         }
 
         Log.debug("Loading password for network \(network.ssid)")
-        return password
+        return try? JSONDecoder().decode(NetworkAuth.self, from: jsonData)
+    }
+
+    func getSavedNetworks() -> [NetworkInfo?] {
+        return keychain.allKeys().map { ssid in
```

Comment:
If you change `map` to `compactMap` the result will be a proper `[NetworkInfo]`, getting rid of the optionality